erpnext/stock/doctype/material_request/material_request.py

```diff
@@ -269,7 +272,10 @@ def update_status(name, status):
 	material_request.update_status(status)
 
 @frappe.whitelist()
-def make_purchase_order(source_name, target_doc=None):
+def make_purchase_order(source_name, target_doc=None, args={}):
```

Comment:
mutable args will cause problems down the line. 

Also get_mapped_doc api expects only one positional argument and one optional argument. This is breaking all get mapped doc. Reference: https://github.com/frappe/erpnext/issues/27557

Reply:
@ankush Linter should have caught this. 

Added [CI check for the same here](https://github.com/ChillarAnand/erpnext/commit/af03a3606dd9a1f002bb5f5e892fa992c7fc32f5). Will raise a PR once [this PR gets merged](https://github.com/frappe/erpnext/pull/27362).

Reply:
Semgrep catches it only when it's actually dangerous. ;)

Reply:
https://github.com/returntocorp/semgrep-rules/blob/develop/python/lang/correctness/common-mistakes/default-mutable-list.yaml 
https://github.com/returntocorp/semgrep-rules/blob/develop/python/lang/correctness/common-mistakes/default-mutable-dict.yaml